src/viam/sdk/CMakeLists.txt

```diff
@@ -0,0 +1,204 @@
+# Declare the viamcpp library, and attach sources,
+# headers for installation
+add_library(viamsdk)
+
+# TODO: This is analogized from
+# https://gitlab.kitware.com/cmake/cmake/-/tree/master/Help/guide/importing-exporting/MathFunctionsComponents
+# but it doesn't really seem to do anything. Everything works fine if
+# it is removed, and naming the alias something other than `viamcpp`
+# doesn't make it available under that alias anyway, as one might
+# hope. What is it supposed to do?
+add_library(viam-cpp-sdk::viamsdk ALIAS viamsdk)
+
+# This bit of magic is required so that we get the public build-time
+# include path of viamapi in *ahead* of the path anchored to `src` by
+# `BASE_DIRS` below, because otherwise we will pull in the static
+# versions of the proto files even if we are generating dynamically.
+target_include_directories(viamsdk
+  PUBLIC
+    "$<BUILD_INTERFACE:$<TARGET_PROPERTY:viam-cpp-sdk::viamapi,INTERFACE_INCLUDE_DIRECTORIES>>"
+)
+
+target_sources(viamsdk
+  PRIVATE
+    # TODO(RSDK-1742): we have to put `registry` up top here because we need the
+    # registry types to be defined first, before anything tries to init them.
+    # this obviously isn't great. it breaks up stylistic ordering and is brittle
+    # when we need to add updates. we should refactor to make this unnecessary.
+    # consider making all necessary runtime values a single `context` that has to
+    # be initialized within main before anything else happens?
+    registry/registry.cpp
+    common/proto_type.cpp
+    common/utils.cpp
+    components/camera/camera.cpp
+    components/camera/client.cpp
+    components/camera/server.cpp
+    components/component_base.cpp
+    components/generic/client.cpp
+    components/generic/generic.cpp
+    components/generic/server.cpp
+    components/motor/motor.cpp
+    components/motor/client.cpp
+    components/motor/server.cpp
+    config/resource.cpp
+    module/handler_map.cpp
+    module/module.cpp
+    module/service.cpp
+    referenceframe/frame.cpp
+    resource/resource.cpp
+    resource/resource_base.cpp
+    resource/resource_manager.cpp
+    resource/resource_server_base.cpp
+    resource/resource_type.cpp
+    robot/client.cpp
```

Comment:
The add of `robot/service.cpp` got lost in a rebase somewhere I think.

Reply:
Entirely possible. The rebases were pretty much manual for this file. Added back.